link/src/main/java/com/stripe/android/link/ui/wallet/WalletUiState.kt

```diff
@@ -67,6 +69,29 @@ internal data class WalletUiState(
         )
     }
 
+    fun updateWithSetDefaultResult(response: ConsumerPaymentDetails): WalletUiState {
+        val updatedPaymentMethod = response.paymentDetails.single()
```

Comment:
My understanding is that this endpoint always returns a list containing the item that we have updated, right?

Reply:
I believe so, but if it doesn't it would crash here. How about doing this as a `.map` on line 255 below, so any error will be handled, and this method receives a single item?

Reply:
Hmm, we’d still need to use `mapCatching` to catch the exception. Maybe I just use `response.paymentDetails.firstOrNull { it.id == paymentMethodIdBeingUpdated }` to be safe against any issues?

Reply:
Yeah, I meant `mapCatching`. That works too, but wouldn't show any error.